test/test_xla_sharding.py

```diff
@@ -103,6 +103,27 @@ def test_deep_copy(self):
         torch_xla._XLAC._get_xla_sharding_spec(xt2))
 
 
+class VirtualDeviceTest(XlaShardingTest):
+
+  @classmethod
+  def setUpClass(cls):
+    os.environ["XLA_USE_SPMD"] = "1"
+    super().setUpClass()
+
+  def test_mark_sharding(self):
+    partition_spec = (0, 1)
+    xt1 = torch.tensor([[1, 2, 3, 4, 5, 6, 7, 8]],
+                       dtype=torch.float,
+                       device=xm.xla_device())
+    xs.mark_sharding(xt1, self._get_mesh((1, self.n_devices)), partition_spec)
+    self.assertTrue(
+        torch.allclose(
+            xt1 + 0,
```

Comment:
Hey @steventk-g could we actually replace this with t2 = t1 + 1 and assert on t2?

Reply:
TODO: adding more unit tests, except the one with a known issue.

Reply:
Added some tests here
https://github.com/pytorch/xla/pull/4214
https://github.com/pytorch/xla/pull/4215

Opened an issue
https://github.com/pytorch/xla/issues/4220